Implementation/Services/index.md

```diff
@@ -0,0 +1,510 @@
+{
+        public interface ISiteHelperService
+    {
+        IPublishedContent GetNewsSection();
+        IPublishedContent GetContactUsPage();
+    }
+}
+```
+Create the concrete service class that implements the interface:
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Umbraco.Core.Models.PublishedContent;
+
+namespace Umbraco8.Services
+{
+    public class SiteHelperService : ISiteHelperService
+    {
+        public IPublishedContent GetNewsSection()
+        {
+            //ToDo: implement this!
+            throw new NotImplementedException();
+        }
+        public IPublishedContent GetContactUsPage()
+        {
+            //ToDo: implement this!
+            throw new NotImplementedException();
+        }
+    }
+}
+```
+
+Register the custom service with Umbraco's underlying DI container using an IUserComposer:
+
+```csharp
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco8.Services;
+
+namespace Umbraco8.Composers
+{
+    public class RegisterSiteHelperServiceComposer : IUserComposer
+    {
+        public void Compose(Composition composition)
+        {         
+            composition.Register<ISiteHelperService, SiteHelperService>(Lifetime.Request);          
+        }
+    }
+}
+```
+
+Note we register the service in 'Lifetime.Request' scope - as we'll be making use of the transient UmbracoContext
+
+Now, implementing the service, we can inject Umbraco's core helper IPublishedContentQuery (the same as you would use when writing Umbraco.Content(1234)) into our custom service, to use in order to locate the special site pages we are after:
+
+```csharp
+using System.Linq;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web;
+
+namespace Umbraco8.Services
+{
+    public class SiteHelperService : ISiteHelperService
+    {
+        private readonly IPublishedContentQuery _contentQuery;
+        public SiteHelperService(IPublishedContentQuery contentQuery)
+        {
+            _contentQuery = contentQuery;
+        }
+        public IPublishedContent GetNewsSection()
+        {
+            var siteRoot = _contentQuery.ContentAtRoot().FirstOrDefault();
+            var newsSection = siteRoot.Children(f => f.ContentType.Alias == "newsSection").FirstOrDefault();
+            return newsSection;
+        }
+        public IPublishedContent GetContactUsPage()
+        {
+            var siteRoot = _contentQuery.ContentAtRoot().FirstOrDefault();
+            var contactUs = siteRoot.Children(f => f.ContentType.Alias == "contactUs").FirstOrDefault();
+            return contactUs;
+        }
+    }
+}
+```
+
+:::Note
+If we planned to use our SiteHelperService outside of a request where UmbracoContext is not available we'd use the UmbracoContextFactory + EnsureUmbracoContext() methods described above to access the content cache.
+:::
+
+:::Note
+Check if _contentQuery.SingleAtXPath("root/homePage/contactUs") is faster... (it would be in V7)
```

Comment:
As noted above - xpath is slower in v8.

Reply:
thought so... any circumstances where it would be better to steer people to use XPath, for these methods?